Maven compatibility issues

The ESH code formatter sets the max line width to 120 while the LineLengthCheck configured in this tool for CheckStyle flags lines over 80 characters as priority 1.

* What is the desired max line length for ESH & openHAB?
* The formatter and the CheckStyle rule should be aligned.

It looks like there is a different problem. I was trying to put together a PR to fix the line length and I realized what @svilenvul mentioned - line length is not one of the checks in the rules.xml.

So digging into this some more, I found that the rules.xml is not being used at all. Instead, Checkstyle is executing sun_checks.xml (appears to be the default). It looks like the settings from the plugin are not executing, but the the goals are - it runs Checkstyle, FindBugs, and PMD. 

Because the properties are not being used, it is dropping the checkstyle-result.xml in target/ rather than target/code-analysis/ so the merge fails and report.html is zero bytes. If I manually copy the checkstyle-result.xml to the code-analysis folder and run again without 'clean' it will generate the report, but it is of course using the default rules.

Originally I was executing as
```
mvn clean install -P check
```

But in response to @kaikreuzer responding to my forum post [here](https://community.openhab.org/t/static-code-analysis-tool/24489/45) I just did a normal

```
mvn clean install
```

and I got the same result.

I then checked out master from the openhab2-addons repo and ran 'mvn clean install' on the Chromecast binding with the same result:

```
$ mvn clean install
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO] Fetching p2.index from http://download.eclipse.org/smarthome/updates-stable/
[INFO] Adding repository http://download.eclipse.org/smarthome/updates-stable
[INFO] Fetching p2.index from https://dl.bintray.com/openhab/p2/openhab-core/2.1.x/ (172B)
[INFO] Fetching p2.index from https://dl.bintray.com/openhab/p2/openhab-core/2.1.x/ (172B)
[INFO] Adding repository https://dl.bintray.com/openhab/p2/openhab-core/2.1.x
[INFO] Adding repository https://dl.bintray.com/openhab/p2/openhab-deps-repo/1.0.10
[INFO] Resolving dependencies of MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO] Resolving class path of MavenProject: org.openhab.binding:org.openhab.binding.chromecast:2.1.0-SNAPSHOT @ /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Chromecast Binding 2.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ org.openhab.binding.chromecast ---
[INFO] Deleting /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:build-qualifier (default-build-qualifier) @ org.openhab.binding.chromecast ---
[INFO] The project's OSGi version is 2.1.0.201704040041
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-id (default-validate-id) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:validate-version (default-validate-version) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:resources (default-resources) @ org.openhab.binding.chromecast ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/src/main/resources
[INFO] 
[INFO] --- tycho-compiler-plugin:1.0.0:compile (default-compile) @ org.openhab.binding.chromecast ---
[INFO] Compiling 4 source files to /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/classes
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:compile (default) @ org.openhab.binding.chromecast ---
[INFO] Changes detected - recompiling the module!
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:testResources (default-testResources) @ org.openhab.binding.chromecast ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/src/test/resources
[INFO] 
[INFO] --- target-platform-configuration:1.0.0:target-platform (default-target-platform) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- tycho-packaging-plugin:1.0.0:package-plugin (default-package-plugin) @ org.openhab.binding.chromecast ---
[INFO] Building jar: /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- tycho-p2-plugin:1.0.0:p2-metadata-default (default-p2-metadata-default) @ org.openhab.binding.chromecast ---
[INFO] 
[INFO] --- static-code-analysis:0.0.3:checkstyle (default) @ org.openhab.binding.chromecast ---
[INFO] Adding dependency to checkstyle:7.2
[INFO] There are 161 errors reported by Checkstyle 7.2 with sun_checks.xml ruleset.
[WARNING] Unable to locate Source XRef to link to - DISABLED
[INFO] 
[INFO] --- static-code-analysis:0.0.3:pmd (default) @ org.openhab.binding.chromecast ---
[WARNING] Unable to locate Source XRef to link to - DISABLED
[WARNING] Unable to locate Source XRef to link to - DISABLED
[INFO] 
[INFO] --- static-code-analysis:0.0.3:findbugs (default) @ org.openhab.binding.chromecast ---
[INFO] Adding dependency to findbugs:3.0.1
[INFO] Adding dependency to bug-pattern:1.2.4
[INFO] Fork Value is true
     [java] Warnings generated: 2
[INFO] Done FindBugs Analysis....
[INFO] 
[INFO] --- static-code-analysis:0.0.3:report (default) @ org.openhab.binding.chromecast ---
Error 
  java.io.FileNotFoundException:
  /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/checkstyle-result.xml (No such file or directory)
[ERROR] java.io.FileNotFoundException: /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/checkstyle-result.xml (No such file or directory)
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
Error 
  SXXP0003: Error reported by XML parser: Premature end of file.
[ERROR] org.xml.sax.SAXParseException; Premature end of file.
[Fatal Error] :-1:-1: Premature end of file.
[WARNING] Cann't select /sca/file/message nodes from /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/f48f91d0_SecondMerge.xml. Empty NodeList will be returned.
org.xml.sax.SAXParseException; Premature end of file.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:205)
	at org.openhab.tools.analysis.report.ReportUtility.selectNodes(ReportUtility.java:346)
	at org.openhab.tools.analysis.report.ReportUtility.appendToSummary(ReportUtility.java:297)
	at org.openhab.tools.analysis.report.ReportUtility.execute(ReportUtility.java:194)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:153)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
[INFO] Empty report will not be appended to the summary report.
[Fatal Error] :-1:-1: Premature end of file.
[WARNING] Cann't select /sca/file/message[@priority=1] nodes from /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/code-analysis/f48f91d0_SecondMerge.xml. Empty NodeList will be returned.
org.xml.sax.SAXParseException; Premature end of file.
	at org.apache.xerces.parsers.DOMParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.DocumentBuilderImpl.parse(Unknown Source)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:205)
	at org.openhab.tools.analysis.report.ReportUtility.selectNodes(ReportUtility.java:346)
	at org.openhab.tools.analysis.report.ReportUtility.checkForErrors(ReportUtility.java:262)
	at org.openhab.tools.analysis.report.ReportUtility.execute(ReportUtility.java:199)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:318)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:153)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:414)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:357)
[INFO] 
[INFO] --- maven-install-plugin:2.3.1:install (default-install) @ org.openhab.binding.chromecast ---
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.jar
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/pom.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT.pom
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/p2content.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT-p2metadata.xml
[INFO] Installing /home/moyerg/workspaces/eclipse-openhab2/openhab2-addons/addons/binding/org.openhab.binding.chromecast/target/p2artifacts.xml to /home/moyerg/.m2/repository/org/openhab/binding/org.openhab.binding.chromecast/2.1.0-SNAPSHOT/org.openhab.binding.chromecast-2.1.0-SNAPSHOT-p2artifacts.xml
[INFO] 
[INFO] --- tycho-p2-plugin:1.0.0:update-local-index (default-update-local-index) @ org.openhab.binding.chromecast ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 17.319s
[INFO] Finished at: Mon Apr 03 20:41:51 EDT 2017
[INFO] Final Memory: 56M/524M
[INFO] ------------------------------------------------------------------------
```
In particular, notice this line:
```
[INFO] There are 161 errors reported by Checkstyle 7.2 with sun_checks.xml ruleset.
```

--- a/src/main/java/org/openhab/tools/analysis/tools/AbstractChecker.java
+++ b/src/main/java/org/openhab/tools/analysis/tools/AbstractChecker.java
@@ -114,7 +114,7 @@ public abstract class AbstractChecker extends AbstractMojo {
             }
         }
 
-        Properties userProps = mavenSession.getUserProperties();
+        Properties userProps = mavenProject.getProperties();
 
         Enumeration<?> e = properties.propertyNames();
         while (e.hasMoreElements()) {

